common/static/common/js/discussion/utils.js

```diff
@@ -321,7 +321,8 @@
             imageUploadUrl = this.urlFor('upload');
             _processor = function(self) {
                 return function(text) {
-                    return self.postMathJaxProcessor(text);
+                    // HTML returned by Markdown is assumed to be safe to render
+                    return self.postMathJaxProcessor(edx.HtmlUtils.HTML(text)).toString();
```

Comment:
Thinking some more about this, I'm now confused. Why was it failing? Adding ``HtmlUtils.HTML`` would be a no-op if the original input was a string. Was it that the value of ``text`` was actually an HTML snippet? If so, maybe this should be:

    return self.postMathJaxProcessor(edx.HtmlUtils.ensureHtml(text).toString());